helix-view/src/view.rs

```diff
@@ -121,6 +121,35 @@ impl View {
         }
     }
 
+    pub fn is_cursor_in_view(&mut self, doc: &Document, scrolloff: usize) -> bool {
```

Comment:
Hmm, this duplicates a lot of the code in `ensure_cursor_in_view` but I don't have ideas on how to refactor it out.

Reply:
I extract calulate login.